### PR TITLE
Create new command 'revokeme <priv>'

### LINKED ISSUE #9576: Add command revokeme <priv>
## Problem

When revoking privileges to yourself, it can take a little long to type the command to revoke a privilege.
I'd like to add a `revokeme` command to make it shorter and quicker to revoke privileges to yourself.

## Solutions

Create a new command (like `grantme`) that revokes an specified privilege to yourself.

## Alternatives

`revoke` is an alternative, however it is longer than `revokeme`.

### PR DESCRIPTION
**Pull request by Panquesito7**
Add compact, short information about your PR for easier understanding:

- Goal of the PR:
   - Make it shorter and quicker to revoke privileges from yourself.
- How does the PR work?
   - Revokes an specified privilege from yourself.
- Does it resolve any reported issue?
  - Yes, closes #9576.

## To do

This PR is Ready for Review.

## How to test

Revoke a privilege to yourself (such as `teleport`) using `revokeme`.

### CODE REVIEW

File: builtin/game/chat.lua [hunk 1]
```diff
@@ -1,27 +1,41 @@
 -- Minetest: builtin/game/chat.lua
 
+-- Helper function that implements search and replace without pattern matching
+-- Returns the string and a boolean indicating whether or not the string was modified
+local function safe_gsub(s, replace, with)
```
Comment (SmallJoker) [inline builtin/game/chat.lua:5]: Unrelated changes. Please undo.

File: builtin/game/chat.lua [hunk 2]
```diff
@@ -224,57 +239,77 @@ core.register_chatcommand("grantme", {
...
+		end
+	end
+
+	for priv, _ in pairs(revokeprivs) do
+		-- call the on_revoke callbacks
+		core.run_priv_callbacks(revokename, priv, caller, "revoke")
+	end
+
+	core.set_player_privs(revokename, privs)
+		core.log("action", caller..' revoked ('
```
Comment (SmallJoker) [inline builtin/game/chat.lua:278]: Misaligned code block.

File: builtin/game/chat.lua (hunk 1 above)
Comment (Panquesito7) [reply builtin/game/chat.lua:5]: My fork of Minetest wasn't updated, so, I just copy-paste current `chat.lua` into my fork.
This is why Git shows unrelated changes here.

File: builtin/game/chat.lua [hunk 3]
```diff
@@ -224,57 +239,77 @@ core.register_chatcommand("grantme", {
...
+
+	for priv, _ in pairs(revokeprivs) do
+		-- call the on_revoke callbacks
+		core.run_priv_callbacks(revokename, priv, caller, "revoke")
+	end
+
+	core.set_player_privs(revokename, privs)
+	core.log("action", caller..' revoked ('
+				..core.privs_to_string(revokeprivs, ', ')
+				..') privileges from '..revokename)
```
Comment (sfan5) [inline builtin/game/chat.lua:280]: too much indentation

File: builtin/game/chat.lua (hunk 1 above)
Comment (SmallJoker) [reply builtin/game/chat.lua:5]: Still not fixed.

File: builtin/game/chat.lua [hunk 4]
```diff
@@ -987,6 +1023,7 @@ core.register_chatcommand("clearobjects", {
 		core.clear_objects(options)
 		core.log("action", "Object clearing done.")
 		core.chat_send_all("*** Cleared all objects.")
+		return true
```
Comment (SmallJoker) [inline builtin/game/chat.lua:1026]: Probably unrelated too.

Comment (sfan5) [reply builtin/game/chat.lua:1026]: Since these changes are already in master they should disappear when merging the PR.
It would be better to remove them from it, though.

Comment (Panquesito7) [reply builtin/game/chat.lua:1026]: So, should I revert it or keep it as-is?

**Comment by ClobberXD** [comment]
Concept makes sense :+1:

File: builtin/game/chat.lua (hunk 4 above)
Comment (ClobberXD) [reply builtin/game/chat.lua:1026]: As sfan5 said, these changes are already in master from various other commits. Removing these extraneous changes would make sure the PR only contains your changes.

Comment (Panquesito7) [reply builtin/game/chat.lua]: Done.

Comment (ClobberXD) [reply builtin/game/chat.lua]: Yes, I understood. I've been there quite a few times as well.

BTW, there are still a lot of extraneous changes that haven't been reverted. All changes up to around L30, for example.

File: builtin/game/chat.lua [hunk 5]
```diff
@@ -6,22 +6,25 @@
...
 	local error_str = "Invalid chat message format - missing %s"
-	local i
+	local str = core.settings:get("chat_message_format")
+	local replaced
 
-	str, i = str:gsub("@name", name, 1)
-	if i == 0 then
+	-- Name
+	str, replaced = safe_gsub(str, "@name", name)
+	if not replaced then
```
Comment (sfan5) [inline builtin/game/chat.lua:15]: changes are still here?

File: builtin/game/chat.lua [hunk 6]
```diff
@@ -224,57 +224,77 @@ core.register_chatcommand("grantme", {
...
+		return false, "Player " .. revokename .. " does not exist."
+	end
+
+	local revokeprivs = core.string_to_privs(revokeprivstr)
+	local privs = core.get_player_privs(revokename)
+	local basic_privs =
+		core.string_to_privs(core.settings:get("basic_privs") or "interact,shout")
+	for priv, _ in pairs(revokeprivs) do
+		if not basic_privs[priv] and
+				not core.check_player_privs(caller, {privs=true}) then
```
Comment (sfan5) [inline builtin/game/chat.lua:243]: can be shortened to: `if not basic_privs[priv] and not caller_privs.privs then`
